Library/Source/NativeEngine.cpp

```diff
@@ -1042,35 +1063,79 @@ namespace babylon
     void NativeEngine::SetTextureSampling(const Napi::CallbackInfo& info)
     {
         const auto textureData = info[0].As<Napi::External<TextureData>>().Data();
-        const auto filter = static_cast<Filter>(info[1].As<Napi::Number>().Uint32Value());
-
-        // STUB: Stub.
+        auto filter = static_cast<uint32_t>(info[1].As<Napi::Number>().Uint32Value());
+
+        constexpr uint32_t filterCount = 12;
+        constexpr std::array<uint32_t, filterCount> bgfxFiltering = {
+            BGFX_SAMPLER_MAG_POINT | BGFX_SAMPLER_MIN_POINT,                            /** nearest is mag = nearest and min = nearest and mip = linear */
+            BGFX_SAMPLER_MIP_POINT,                                                     /** Bilinear is mag = linear and min = linear and mip = nearest */
+            0,                                                                          /** Trilinear is mag = linear and min = linear and mip = linear */
+            BGFX_SAMPLER_MIN_POINT | BGFX_SAMPLER_MAG_POINT | BGFX_SAMPLER_MIP_POINT,   /** mag = nearest and min = nearest and mip = nearest */
+            BGFX_SAMPLER_MAG_POINT | BGFX_SAMPLER_MIP_POINT,                            /** mag = nearest and min = linear and mip = nearest */
+            BGFX_SAMPLER_MAG_POINT,                                                     /** mag = nearest and min = linear and mip = linear */
+            BGFX_SAMPLER_MAG_POINT,                                                     /** mag = nearest and min = linear and mip = none */
+            BGFX_SAMPLER_MAG_POINT | BGFX_SAMPLER_MIN_POINT,                            /** mag = nearest and min = nearest and mip = none */
+            BGFX_SAMPLER_MIP_POINT | BGFX_SAMPLER_MIP_POINT,                            /** mag = linear and min = nearest and mip = nearest */
+            BGFX_SAMPLER_MIN_POINT,                                                     /** mag = linear and min = nearest and mip = linear */
+            0,                                                                          /** mag = linear and min = linear and mip = none */
+            BGFX_SAMPLER_MIN_POINT };                                                   /** mag = linear and min = nearest and mip = none */
+        filter = std::min(filter, filterCount - 1);
```

Comment:
This function is called from JS and we have no guarantee that it will be called with a correct value. I don't want the app the crash because the user sets an invalid value as a filter value.

Reply:
I think you should check this from the JS side and throw a JS exception.